src/components/Session/Deposit.jsx

```diff
@@ -0,0 +1,65 @@
+const React = window.React = require('react');
+import DepositCurrency from './Deposit/DepositCurrency.jsx';
+import DepositAnchors from './Deposit/DepositAnchors.jsx';
+
+export default class Deposit extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      'selectedAsset': null
+    }
+  }
+
+  _onCurrencyChange(newAsset) {
+    this.setState({ 'selectedAsset': newAsset });
+  }
+
+  render() {
+    const account = this.props.d.session.account;
+    const allAssets = account.getSortedBalances({hideNative: true}); // From MagicSpoon.Account
+
+    let currencyPicker;
+    let anchors;
+    if (allAssets.length == 0) {
+      const noTrustLines = (<div>
+        You haven't trusted any assets. Click here to <a href="#account/addTrust">create your first trust line</a>.
+        </div>);
+      currencyPicker = noTrustLines;
+      anchors = noTrustLines;
+    } else {
+      currencyPicker = (<DepositCurrency
+        d={this.props.d}
+        allAssets={allAssets}
+        selectedAsset={this.state.selectedAsset || allAssets[0].code}
+        onCurrencyChange={this._onCurrencyChange.bind(this)}
```

Comment:
Simpler and easier to maintain

```js
onCurrencyChange={newAsset => this.setState({'selectedAsset': newAsset})}
```

Reply:
Also, note that theres no need for bind since fat arrow automatically does it

Reply:
ok, will update